leasing/migrations/0076_paikkatietovipunen_vuokraalueet_view_vuokraustunnus_and_tyypin_tunnus.sql

```diff
@@ -0,0 +1,287 @@
+                FROM leasing_tenantcontact tc
+                WHERE tc.type::text = 'contact'::text
+                    AND (
+                        tc.end_date IS NULL
+                        OR tc.end_date > now()
+                    ) AND (
+                        tc.tenant_id IN (
+                            SELECT t.id
+                            FROM leasing_tenant t
+                            WHERE t.lease_id = lease.id
+                        )
+                    )
+            )
+        )
+    ) AS yhteyshenkilot_osoitteet,
+    (
+        SELECT aa.address
+        FROM leasing_leaseareaaddress aa
+        WHERE aa.lease_area_id = l_area.id AND aa.is_primary
+        LIMIT 1
+    ) AS ensisijainen_osoite,
+    lease.start_date AS vuokraus_alkupvm,
+    lease.end_date AS vuokraus_loppupvm,
+    np.name AS irtisanomisaika,
+    liu.name AS vuokraus_kayttotarkoitus,
+    CASE
+        WHEN lease.end_date IS NULL THEN 'kyllä'::text
+        WHEN lease.end_date > now() THEN 'kyllä'::text
+        WHEN lease.end_date < now() THEN 'ei'::text
+        ELSE NULL::text
+    END AS vuokraus_voimassa,
+    CASE
+        WHEN lease.state::text = 'permission'::text THEN 'Lupa'::text
+        WHEN lease.state::text = 'lease'::text THEN 'Vuokraus'::text
+        WHEN lease.state::text = 'short_term_lease'::text THEN 'Lyhytaikainen vuokraus'::text
+        WHEN lease.state::text = 'long_term_lease'::text THEN 'Pitkäaikainen vuokraus'::text
+        WHEN lease.state::text = 'reservation'::text THEN 'Varaus'::text
+        WHEN lease.state::text = 'reserve'::text THEN 'Varanto'::text
+        WHEN lease.state::text = 'application'::text THEN 'Hakemus'::text
+        WHEN lease.state::text = 'rya'::text THEN 'RYA'::text
+        ELSE NULL::text
+    END AS vuokraus_tila,
+    l_area.area AS pintaala,
+    l_area.section_area AS leikkauspintaala,
+    (
+        SELECT sum(leasing_rent.payable_rent_amount) AS sum
+        FROM leasing_rent
+        WHERE leasing_rent.lease_id = lease.id
+    ) AS perittava_vuokra_summa,
+    (
+        SELECT concat(
+            to_char(
+                leasing_rent.payable_rent_start_date::timestamp with time zone, 'fmDD.fmMM.YYYY'::text
+            ),
+            ' - ',
+            to_char(
+                leasing_rent.payable_rent_end_date::timestamp with time zone, 'fmDD.fmMM.YYYY'::text
+                )
+        ) AS concat
+        FROM leasing_rent
+        WHERE leasing_rent.lease_id = lease.id
+        ORDER BY leasing_rent.payable_rent_start_date DESC
+        LIMIT 1
+    ) AS perittava_vuokra_jakso,
+    lease.note AS vuokraus_huom,
+    ltype.name AS vuokraus_tyyppi,
+    CASE
+        WHEN l_area.location::text = 'surface'::text THEN 'Yläpuolella'::text
+        WHEN l_area.location::text = 'underground'::text THEN 'Alapuolella'::text
+        ELSE NULL::text
+    END AS sijainti,
+    st_transform(l_area.geometry, 3879) AS geometria,
+    CASE
```

Comment:
I suppose the real change is here:
```
+    CASE
+        WHEN su.id = 1 THEN 'MAKE/Tontit'
+        WHEN su.id = 2 THEN 'AKV'
+        WHEN su.id = 3 THEN 'KUVA (LIPA)'
+        WHEN su.id = 4 THEN 'KUVA (UPA)'
+        WHEN su.id = 5 THEN 'KUVA (NUP)'
+        ELSE NULL::text
+    END AS palvelukokonaisuus,
+    CASE
+        WHEN lessor.type::text = 'person'::text THEN concat(lessor.last_name, ' ', lessor.first_name)
+        WHEN lessor.type::text = 'business'::text THEN lessor.name::text
+        WHEN lessor.type::text = 'unit'::text THEN lessor.name::text
+        WHEN lessor.type::text = 'association'::text THEN lessor.name::text
+        WHEN lessor.type::text = 'other'::text THEN lessor.name::text
+        ELSE NULL::text
+    END AS vuokranantaja
```
and also
```
+    LEFT JOIN leasing_contact lessor ON lessor.id = lease.lessor_id
+    LEFT JOIN leasing_serviceunit su ON lease.service_unit_id = su.id
```

Really difficult to read the diff since there's so much formatting changed.

The real additions look good, I did not review any formatting, whether there could be any mistakes.